internal/cephfs/errors.go

```diff
@@ -16,15 +16,45 @@ limitations under the License.
 
 package cephfs
 
-import "errors"
+import (
+	"errors"
+	"fmt"
+)
 
 var (
 	// ErrInvalidVolID is returned when a CSI passed VolumeID is not conformant to any known volume ID
 	// formats.
 	ErrInvalidVolID = errors.New("invalid VolumeID")
 	// ErrNonStaticVolume is returned when a volume is detected as not being
 	// statically provisioned.
-	ErrNonStaticVolume = errors.New("volume not static")
+	ErrNonStaticVolume = errors.New(" volume not static")
+	// ErrVolNotEmpty is returned when a subvolume has snapshots in it.
+	ErrVolNotEmpty = errors.New(" Error ENOTEMPTY")
```

Comment:
@Madhu-1 I was using it, but the problem is that, then we will get linter error saying "Error strings shouldnot be capitiilzed.."
this is the workaround.

Reply:
in that case, we can just define this as a string and use it, I assume these error string are only for comparison with the error that we got from the CLI command. we are again converting back from error to string for comparison, it better to declare as a string and compare it directly.

Reply:
Not really madhu, we do Error comparison in callers or second level callers and it make use of these strings..

Reply:
if am correct theses errors are only used in clone.go volume.go and snapshot.go (which we call the CLI for subvolume operations) not returned to any callers. these errors are converted back to a string for comparison with CLI error output string. instead of defining as errors and converting back the string for comparison. we can avoid the conversion by defining it as strings.

Reply:
with the current codebase, the callers are comparing the errors defined in cephcsi, not the errors which represent the CLI errors. IMO we  need to keep both separate

Reply:
@Madhu-1 we are making use of these errors in ControllerServer too, thats why I said, in 2 levels .Also we make use of error unwrapping check with `errros.Is(...)` for ex: https://github.com/ceph/ceph-csi/pull/394/files#diff-15646e8d8f257046456241d1496ced69R93. Also we are not converting back to string and validating at places , for example https://github.com/ceph/ceph-csi/pull/394/files#diff-67a5de1fe7bbb6359e176b0c6b3222ccR123 . So imo, keeping it in error form is good enough. We do some string comparison on leaf function, but thats fine I believe. Also changing again everything back to `string` and in the callers I can do later too or in follow-up if really required. wdyt?

Reply:
we are defining the strings at https://github.com/ceph/ceph-csi/blob/74ba85f87b1526e615f1643d3475316137a18264/internal/cephfs/volume.go#L38-L42 these string messages are only used for comparison with the CLI error messages and it's not returned to the user. and we have defined error messages at https://github.com/ceph/ceph-csi/blob/74ba85f87b1526e615f1643d3475316137a18264/internal/cephfs/errors.go#L24-L29 which are returned to the users and which can be easily understood by the user. here are we removing the string we defined for the comparison and creating only one error which is used for both comparison and returning to the user (with hack (leaving first value as blank in errors.New*()` . I would suggest keeping the error messages and error string separate as it was before

Reply:
@Madhu-1 no more strings in volume.go , I aleady got rid of it in this PR sometime back.. may be you are in different branch?

Reply:
Also, no hacks with the space now, everything is `fmt.Errorf()`

Reply:
I am not happy with this approach :-1: we should not return ceph specific error messages to the user. Earlier implementation was much better

Reply:
sure @Madhu-1 . Let me mark it in the follow up tracker to revisit . Thanks @Madhu-1 :+1: